spec/support/shared_examples/expand_resources_includes.rb

```diff
@@ -0,0 +1,25 @@
+RSpec.shared_context "for 'expand' query optimizations" do
+  context "expand attribute query optimizations" do
+    it "removes N+1's from the index query for subcollections/virtual_attributes" do
+      api_basic_authorize action_identifier(resource, :read, :resource_actions, :get)
+
+      expands    = ["resources"]
+      from_match = [resource]
+
+      if defined?(uses)
+        from_match << uses
+      end
+
+      attrs                = defined?(attributes) ? attributes : "resources"
+      query_match          = /SELECT.*FROM\s"(?:#{from_match.join("|")})"/m
+      expected_query_count = defined?(query_count) ? query_count : 3
```

Comment:
Can you just add a comment on why the magic number 3 (or perhaps move into a named constant?)

Reply:
Sure, I will do both.

Reply:
Added the comment and made it a `let` attribute.